### PR TITLE
Revert "Portability fixes"

### PR DESCRIPTION
**Pull request by endoffile78**
This reverts commit f3469070fe899e8e4fd88665386a55bad9f77cd8.

---
This change is [Reviewable](https://reviewable.io/reviews/toktok/c-toxcore/395)

### CODE REVIEW

**Comment by iphydf** [comment]
:lgtm_strong:

---

Reviewed 27 of 27 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/toktok/c-toxcore/395#-:-K_kV9HpsXHhPcSgkZj8:b-jbsf8z)*

File: CMakeLists.txt [hunk 1]
```diff
@@ -56,21 +56,22 @@ macro(add_flag flag)
...
-
-# Error on non-ISO C.
-add_cflag("-pedantic-errors")
 
 option(WARNINGS "Enable additional compiler warnings" ON)
 if(WARNINGS)
+  # Set standard version for compiler.
+  add_cflag("-std=gnu99")
+  add_cxxflag("-std=c++98")
+
```
Comment (cebe) [inline CMakeLists.txt]: this looks incorrect to me. You only define the C standard if warnings are enabled?

Comment (iphydf) [inline CMakeLists.txt]: Yes. By default, gcc uses gnu89, which is sufficient for us at the moment (soon it won't be). @Zer0-One will hopefully make the original PR again which reverts this revert.

Comment (jin-eld) [inline CMakeLists.txt]: Just curious, why are cxxflags set in a C-only library?

Comment (iphydf) [inline CMakeLists.txt]: We also compile the code as C++ to make sure it's C++ compatible. Reasons:
- Additional C++-only tooling becomes available (static analysers).
- A few checks performed by C++ compilers (enum compatibility, void pointer casts) are useful for us.

**Comment by nurupo** [comment]
@endoffile78 Would be nice if the PR mention why the commit was reverted. If I remember right from the IRC discussions (which not everyone going over PRs follows closely), it's because that commit broke toxcore on OS X?

**Comment by Zer0-One** [comment]
@nurupo correct

**Pull request merged**